src/ansys/dynamicreporting/core/utils/report_remote_server.py

```diff
@@ -882,34 +883,81 @@ def export_report_as_pdf(
         delay=None,
         exec_basis=None,
         ansys_version=None,
+        image_size=False,
     ):
         if query is None:
             query = {}
         query["print"] = "pdf"
         url = self.build_url_with_query(report_guid, query)
-        file_path = os.path.abspath(file_name)
-        if has_qt and (parent is not None):
-            from .report_download_pdf import NexusPDFSave
-
-            app = QtGui.QGuiApplication.instance()
-            worker = NexusPDFSave(app)
-            _ = worker.save_page_pdf(url, filename=file_path, page=page, delay=delay)
-            return
+        # If image_size is set to True, set width and height to make sure
+        # only 2 images are printed per page. Needed for the Fluent export
+        # capability
+        if image_size:  # pragma: no cover
+            root_report = self.get_object_from_guid(report_guid)
+            reports_to_modify = {root_report: copy.deepcopy(root_report.get_property())}
+            root_report.add_property({"width": 800, "height": 600})
+            # This property will propagate to all the templates. Reset it to zero on the
+            # Logo template so the Logo image doesn't get resized.
+            # Ideally you want to walk the tree from root_report down and find the Logo
+            # templates only there. In practice, that's too time consuming so just
+            # modify all Logo templates in the database and then reset them.
```

Comment:
@margalva 
So modifying *ALL* Logo templates is much more expensive than walking the root report. The get_objects call can be pretty expensive. I would just check the concerned report only. Modifying all Logo templates in the database seems unnecessary.
Also I am not completely sure about the need for this kind of a hack and maybe I can be more helpful if I do, but the overall approach seems pretty hacky to me. For example - things like checking for the string "Logo" and assuming it is nothing but a Logo image, the hardcoded values for width and height 800x600, etc. If this heavily customized based on Fluent's requests as mentioned in the comment above, I would come up with a generic fix and avoid this one.
But nonetheless, I think @randallfrank , the original author of this, must have a look. There should be a cleaner way, eg. via Qt itself possibly.

Reply:
@viseshrp see your team messages for a more detailed discussion of this hack.
Yes this is a hack to quickly fix an issue they currently have - no reason to make a more extensive change since we're going to change all of this functionality soon anyway. Qt addresses this issue via using the footers, which Fluent is going to introduce via adding footer templates, but that alone isn't enough if they don't fix the image sizes.
I tested the Logo modifications vs. searching the Logo template explicitly in the whole tree and building the tree is way more expensive, considering how complex their final template is. See first commit with the tree building function, it takes significantly longer than getting all the Logo templates out as it involves recursively using the get_objects call.